src/Access/SettingsProfileElement.cpp

```diff
@@ -236,6 +253,81 @@ std::vector<UUID> SettingsProfileElements::toProfileIDs() const
     return res;
 }
 
+
+void SettingsProfileElements::normalize()
+{
+    /// Ensure that each element represents either a setting or a profile.
+    {
+        SettingsProfileElements new_elements;
+        for (auto & element : *this)
+        {
+            if (element.parent_profile && !element.setting_name.empty())
+            {
+                SettingsProfileElement new_element;
+                new_element.parent_profile = element.parent_profile;
+                element.parent_profile.reset();
+                new_elements.push_back(std::move(new_element));
+            }
+        }
+        insert(end(), new_elements.begin(), new_elements.end());
+    }
+
+    /// Partitioning: first profiles, then settings.
+    /// We use std::stable_partition() here because we want to preserve the relative order of profiles and the relative order of settings.
+    /// (We need that order to be preserved to remove duplicates correctly - see below.)
+    auto profiles_begin = begin();
+    auto profiles_end = std::stable_partition(begin(), end(), [](const SettingsProfileElement & element) { return static_cast<bool>(element.parent_profile); });
```

Comment:
I understand the idea behind this, but it still feels wrong that we have two different entities (settings and profiles) in one vector and do this tricky reordering each time. Maybe we should separate it somehow in the first place?